linux/LinuxMachine.c

```diff
@@ -820,3 +824,108 @@ bool Machine_isCPUonline(const Machine* super, unsigned int id) {
    assert(id < super->existingCPUs);
    return this->cpuData[id + 1].online;
 }
+
+static int humanTimeUnit(char* buffer, size_t size, unsigned long long int value) {
+
+   if (value < 1000)
+      return xSnprintf(buffer, size, "%3lluns", value);
+
+   if (value < 10000)
+      return xSnprintf(buffer, size, "%1llu.%1lluus", value / 1000, (value % 1000) / 100);
+
+   value /= 1000;
+
+   if (value < 1000)
+      return xSnprintf(buffer, size, "%3lluus", value);
+
+   if (value < 10000)
+      return xSnprintf(buffer, size, "%1llu.%1llums", value / 1000, (value % 1000) / 100);
+
+   value /= 1000;
+
+   if (value < 1000)
+      return xSnprintf(buffer, size, "%3llums", value);
+
+   if (value < 10000)
+      return xSnprintf(buffer, size, "%1llu.%1llus", value / 1000, (value % 1000) / 100);
+
+   value /= 1000;
+
+   if (value < 600)
+      return xSnprintf(buffer, size, "%3llus", value);
+
+   value /= 60;
+
+   if (value < 600)
+      return xSnprintf(buffer, size, "%3llum", value);
+
+   value /= 60;
+
+   if (value < 96)
+      return xSnprintf(buffer, size, "%3lluh", value);
+
+   value /= 24;
+
+   return xSnprintf(buffer, size, "%3llud", value);
+}
+
+double Machine_getGpuUsage(const Machine* super, double values[]) {
```

Comment:
I consider this function belongs to Platform.c, not here.
Also, pass the pointer to the whole Meter instance (`Meter *`), not just the `values` array, as the Meter instance would be more useful.

Reply:
Given the CPU data is also handled as part of the `*Machine` structure, putting the GPU data there feels much more natural to me, than putting it in the `*Platform` code. This also allows for putting the code in the correct structures more easily, without violating the encapsulation.